src/System.ObjectModel/src/System/Collections/ObjectModel/ObservableCollection.cs

```diff
@@ -84,6 +69,29 @@ public void Move(int oldIndex, int newIndex)
         {
             MoveItem(oldIndex, newIndex);
         }
+        
+        /// <summary>
+        /// Adds a range of objects but does not raise the OnCollectionChanged event until all objects are added.
+        /// </summary>
+        /// <param name="collection">The set of objects to be added.</param>
+        /// <param name="clearFirst">Determines whether or not to remove any existing items in the collection first. Defaults to False.</param>
+        public void AddRange(IEnumerable<T> collection, bool clearFirst = false)
+        {
+            CheckReentrancy();
+
+            if (clearFirst) Items.Clear();
+
+            var newIndex = Items.Count;
+            foreach (var item in collection)
+            {
+                base.InsertItem(newIndex, item);
+                newIndex++;
+            }
+
+            OnPropertyChanged(CountString);
+            OnPropertyChanged(IndexerName);
+            OnCollectionChanged(new NotifyCollectionChangedEventArgs(NotifyCollectionChangedAction.Add, new List<T>(collection)));
```

Comment:
It'd be nice if we didn't have to allocate/copy-to a new `List<T>` if the passed-in collection already implements `IList`.